test/unit/test-performance.js

```diff
@@ -687,3 +687,207 @@ describes.realWin('performance with experiment', {amp: true}, env => {
+        startTime: 10,
+      }];
+      const list = {
+        getEntries() {
+          return entries;
+        },
+      };
+      // Fake a triggering of the firstInput event.
+      performanceObserver.triggerCallback(list);
+      expect(perf.events_.length).to.equal(2);
+      expect(perf.events_[0])
+          .to.be.jsonEqual({
+            label: 'fp',
+            delta: 11,
+          },
+          {
+            label: 'fcp',
+            delta: 15,
+          });
+      delete env.win.PerformanceEventTiming;
+    });
+  });
+
+  describe('should forward first input metrics for performance entries', () => {
+    it('created before performance service registered', () => {
+      // Pretend that the EventTiming API exists.
+      env.win.PerformanceEventTiming = true;
+
+      const entries = [{
+        cancelable: true,
+        duration: 8,
+        entryType: 'firstInput',
+        name: 'mousedown',
+        processingEnd: 105,
+        processingStart: 103,
+        startTime: 100,
+      }];
+      const getEntriesByType = env.sandbox.stub();
+      getEntriesByType.withArgs('firstInput').returns(entries);
+      getEntriesByType.returns([]);
+      env.sandbox.stub(env.win.performance, 'getEntriesByType')
+          .callsFake(getEntriesByType);
+
+      installPerformanceService(env.win);
+
+      const perf = Services.performanceFor(env.win);
+
+      expect(perf.events_.length).to.equal(1);
+      expect(perf.events_[0])
+          .to.be.jsonEqual({
+            label: 'fid',
+            delta: 3,
+          });
+
+      delete env.win.PerformanceEventTiming;
+    });
+
+    it('created after performance service registered', () => {
+      // Pretend that the EventTiming API exists.
+      env.win.PerformanceEventTiming = true;
+
+      // Stub and fake the PerformanceObserver constructor.
+      const PerformanceObserverStub = env.sandbox.stub();
+
+      let performanceObserver;
+      PerformanceObserverStub.callsFake(callback => {
+        performanceObserver = new PerformanceObserverImpl(callback);
+        return performanceObserver;
+      });
+      env.sandbox.stub(env.win, 'PerformanceObserver')
+          .callsFake(PerformanceObserverStub);
+
+      installPerformanceService(env.win);
+
+      const perf = Services.performanceFor(env.win);
+
+      const entries = [{
+        cancelable: true,
+        duration: 8,
+        entryType: 'firstInput',
+        name: 'mousedown',
+        processingEnd: 105,
+        processingStart: 103,
+        startTime: 100,
+      }];
+      const list = {
+        getEntries() {
+          return entries;
+        },
+      };
+      // Fake a triggering of the firstInput event.
+      performanceObserver.triggerCallback(list);
+      expect(perf.events_.length).to.equal(1);
+      expect(perf.events_[0])
+          .to.be.jsonEqual({
+            label: 'fid',
+            delta: 3,
+          });
+      delete env.win.PerformanceEventTiming;
```

Comment:
consider put the delete into `afterEach`

Reply:
Hmm - one of the tests fakes `env.win.PerformancePaintTiming` and the three others fake `env.win.PerformanceEventTiming`. We'll introduce other faked properties in this manner (like `window.PerformanceLayoutJank` so I think leaving these in the test cases is okay?

Reply:
Ah, didn't realize they're different.